presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -821,11 +840,14 @@ private static AWSCredentialsProvider getCustomAWSCredentialsProvider(URI uri, C
         private long streamPosition;
         private long nextReadPosition;
 
-        public PrestoS3InputStream(AmazonS3 s3, String host, Path path, int maxAttempts, Duration maxBackoffTime, Duration maxRetryTime)
+        private final boolean requesterPaysEnabled;
```

Comment:
move this to other members that form immutable configuration state. I'd add the new `requesterPaysEnabled` field right after `path`, and before retry related members. Please use same ordering in fields, constructor parameters and constructor assignments.